### PR TITLE
exit early if buffer is empty

### PR DESCRIPTION
**Pull request by tao-stones**
#### Problem

A `warn` of finalized slot isn't found in priority fee cache, is printed for empty slot. This `warn` is false negative, since empty slot will not have any prioritization fee info.

#### Summary of Changes
- exit early if `unfinalized` buffer is empty.

Fixes #

### CODE REVIEW

File: runtime/src/prioritization_fee_cache.rs [hunk 1]
```diff
@@ -293,6 +293,10 @@ impl PrioritizationFeeCache {
         bank_id: BankId,
         metrics: &PrioritizationFeeCacheMetrics,
     ) {
+        if unfinalized.is_empty() {
+            return;
+        }
```
Comment (CriesofCarrots) [inline runtime/src/prioritization_fee_cache.rs:298]: Why does this early return help? Presumably there could be other (newer) slots in the `unfinalized` map when an empty slot is finalized.

Alternate question: are there known times, like on boot, when `finalize_slot` is called but `unfinalized` is empty?

**Review by CriesofCarrots — COMMENTED** [review]
Hmm, let me make sure I understand: a node (falsely) hits the warning when it finalizes a slot that doesn't have any transactions (like an epoch boundary slot with a bunch of rewards, for instance), right?

What are the legitimate triggers for the warning? Just map corruption?

File: runtime/src/prioritization_fee_cache.rs (hunk 1 above)
Comment (tao-stones) [reply runtime/src/prioritization_fee_cache.rs:298]: 1. when a slot is actually empty - no transactions at all, only ticks; the warning will be triggered falsely
2. when a slot has transactions, but none of them set priority fee, so the `unfinalized` cache will be empty, the warning will be triggered falsely too
The early return covers these two cases. 

You brought a good point: "there could be other (newer) slots in the unfinalized map when an empty slot is finalized", it should be a rare case in mnb, do we want to be warned about it? Think about it, probably not necessary - nothing wrong with an slot without priority fee.

Comment (CriesofCarrots) [reply runtime/src/prioritization_fee_cache.rs:298]: Okay cool, thanks for walking through that!

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**